lucene/native/build.gradle

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This gets separated out from misc module into a native module due to incompatibility between cpp-library and java-library plugins.
+ * For details, please see https://github.com/gradle/gradle-native/issues/352#issuecomment-461724948
+ */
+apply plugin: 'cpp-library'
+
+description = 'Module for native code'
+
+library {
+    baseName = 'NativePosixUtil'
+
+    privateHeaders.from file(System.getProperty('java.home') + '/include')
+    privateHeaders.from file(System.getProperty('java.home') + '/include/darwin')
+    privateHeaders.from file(System.getProperty('java.home') + '/../include/solaris')
```

Comment:
Are these really a folder up from java.home? On my linux (java 11-15) they seem to be right under JDK's home (although I install Java manually, not via distribution-specific package manager).

Reply:
These paths were ported based on the original ant target definition below :

https://github.com/apache/lucene-solr/blob/005fec5a58e1993a1ce7ae235060ad3bf2adb9c7/lucene/contrib/misc/build.xml#L44-L45

My guess is these paths might now be deprecated in newer JDK installations. 

Do you think this native build feature should be back ported to prior versions? If so, I think it might be safer to have those paths as well.

Reply:
These paths are wrong, I think. I don't know where they came from but they should correspond to the layout of newer JDK distributions (openjdk). 

I'm not sure Solaris is still needed - I can't test it on Solaris.